src/runtime_src/core/pcie/windows/alveo/device_windows.cpp

```diff
@@ -1275,6 +1365,19 @@ struct function2_getter : QueryRequestType
   }
 };
 
+
+template <typename QueryRequestType, typename Getter>
+struct function4_get : virtual QueryRequestType
```

Comment:
Why `virtual` here? Do you have a diamond problem to solve?
https://en.cppreference.com/w/cpp/language/derived_class

Reply:
Lack of comments prevent me from answering this.  I have no clue why virtual derivation was added, but I originally did this and I am fully aware of what it means, but for the heck of it I cannot see why it is there now, and github logs do not tell me why I added it.  Could have been some interim implementation that was since changed?  Again, my item on my to-do list.